general/management/commands/report_count_statuses.py

```diff
@@ -74,153 +80,129 @@ def handle(self,  *args, **options):
             'User.num_posts': list(),
         }
 
+        # IMPLEMENTATION NOTE: on the code below we iterate multiple times on Sounds, Packs and Users tables.
+        # This is done in this way because using the Django ORM we can't annotate counts from different tables in a
+        # single queryset (annotations that would require SQL JOINs with more than one table). If we do that, we
+        # get wrong results for the annotated fields as the generated SQL duplicates results for each table (see
+        # https://code.djangoproject.com/ticket/10060).
+
         # Sounds
         total = Sound.objects.all().count()
-        print "Iterating over existing %i sounds..." % total,
-
-        annotations = {
-            'real_num_comments': Count('comments'),
-            'real_num_ratings': Count('ratings'),
-            'real_avg_rating': Avg('ratings__rating'),
-        }
-        if not options['skip-downloads']:
-            annotations['real_num_downloads'] = Count('downloads')
-
-        for count, sound in enumerate(Sound.objects.all().annotate(**annotations).iterator()):
-
-            needs_save = False
 
-            # Check num_comments
+        # Look at number of comments
+        for count, sound in enumerate(Sound.objects.all().annotate(real_num_comments=Count('comments')).iterator()):
             real_num_comments = sound.real_num_comments
             if real_num_comments != sound.num_comments:
                 mismatches_report['Sound.num_comments'] += 1
                 mismatches_object_ids['Sound.num_comments'].append(sound.id)
                 sound.num_comments = real_num_comments
-                needs_save = True
-
-            # Check num_downloads
-            if not options['skip-downloads']:
-                real_num_downloads = sound.real_num_downloads
-                if real_num_downloads != sound.num_downloads:
-                    mismatches_report['Sound.num_downloads'] += 1
-                    mismatches_object_ids['Sound.num_downloads'].append(sound.id)
-                    sound.num_downloads = real_num_downloads
-                    needs_save = True
-
-            # Check num_ratings and avg_rating
+                if not options['no-changes']:
+                    sound.is_index_dirty = True
+                    sound.save()
```

Comment:
This is potentially going to be slow (especially the first time you run it again to fix all the counts). Can we do a bulk save (wrap it in a transaction)?